47 resize video window

varun-un opened the pull request:
Resizes window to be a little bit larger, but mainly just adds in a basic 3D visualization for the rocket's orientation on the video window.

jhauerst commented on coders/APRSTelem.js:
I'm not sure what happened to this file, but somehow git thinks the entire file changed when most of it's the same. Can you clarify what changes were made?

jhauerst commented on main.js:
```diff
@@ -880,12 +881,15 @@ if (config.dataDebug.value) {
+              
+              // We're using an empty stateflags array but our improved getStateflag will handle this
+              aprsMsg.stateflagsFormat = []; // Clear existing format if any
+              
```
```findStateflags()```  populates the APRSTelem internal stateflags variable with the correct values. This way the frontend does not have to know how the stateflags are encoded, since it can find them by name. Removing this method is why the ```getStateflag()``` method needed to be changed, and this new code will almost certainly break when trying to use any other stateflags besides stage. If there is a bug that was causing stateflags to break this is probably not the way to fix it.

jhauerst commented on src/libraries/babylon.js:
Since babylon has multiple files, they should be put in their own folder in libraries.

jhauerst commented on src/video/myscript.js:
Not sure what this file is for, but I'm guessing it's a test file that can be removed.

jhauerst commented on src/video/video.js:
```diff
@@ -444,4 +470,90 @@ window.onload = () => {
+
+  createScene().then(scene => {
+      babylonEngine.runRenderLoop(() => scene.render());
+  });
```
It would be nice to only run the render loop if the 3d visualization is being shown.

jhauerst commented on src/video/video.js:
```diff
@@ -349,30 +352,46 @@ window.onload = () => {
-        alt.setAttribute("data-value-text", msg.getAlt());
-        alt.setAttribute("data-value", msg.getAlt() / 1000);
-        document.getElementById("alt-text").textContent = msg.getAlt() + " ft";
+        const altValue = Math.max(msg.getAlt() || 0, 0);
```
I think I agree with Copilot here, the ```|| 0``` doesn't seem to be necessary, but it's also not causing any harm.

varun-un replied:
I forget why exactly, but I added this because I think I was getting issues where .getAlt() wasn't returning numeric values like null or something, so the `|| 0` helped with that

jhauerst commented on src/video/video.html:
```diff
@@ -52,116 +47,113 @@
             data-title="Altitude"
             data-value="0"
-            data-value-text="&#8212;"
+            data-value-text="—" /* Initial value, JS updates. Will be transparent */
```
This is not how you make an HTML comment

jhauerst commented on src/video/video.html:
```diff
@@ -15,14 +15,11 @@
     <title>Video Window</title>
   </head>
   <body>
-    <!--Application top bar-->
```
All the HTML comments are removed, I'm guessing claude did this? They should be brought back because they help keep track of how the HTML is organized.

jhauerst commented on src/index.css:
```diff
@@ -434,6 +436,23 @@ span .value {
+  
+  .gauge-text[data-length="5"] {
+    font-size: calc(2.2vh + 2px); /* Even smaller for 5 digits */
+  }
```
I feel like responsive font sizing for the whole application is a bit out of the scope of this issue, and I don't think there were any major issues before. Is there a specific issue you found with font size on high resolution displays?

jhauerst reviewed (changes requested):
Mainly, I'll need the test files you used to make sure everything works. But also, it seems to me that claude or another LLM made most of these changes directly, and that it doesn't really understand how the ground station works. There's a lot of odd stuff going on (removing HTML comments, reworking stateflags) that's going to need to be fixed before this can be merged.

varun-un commented on coders/APRSTelem.js:
```diff
@@ -1,308 +1,333 @@
+    return this.rawStateflags;
+  }
+
+  getStateflag(flagName) {
```
this is what changed, I think some of the test files was formatted variably to use stateflags and/or stage so it would be like the airbrake and stuff's data for the main window wasn't being pulled in, so this was added

varun-un commented:
> A couple of global comments aside from comments for specific files.
> 
> * The new "models" folder should be moved into the "src" folder, since the only thing that interacts with the models is the frontend.
> * Is there a specific reason you switched everything to using Roboto? I feel like it's way more effort to manually include the font files and stuff compared to just using built in fonts.
> * What data did you use to test the visualization? The default data doesn't include orientation, so could you share the files you used for testing?

[test-0.csv](https://github.com/user-attachments/files/21400863/test-0.csv)
This was the test file, I just added random numbers to test it, but it just looks for Orientation columns.

I switched it to roboto kinda randomly but I think the default font looks meh, I don't mind reverting the roboto but I do think we should change that. The manually included files was to allow it to use the font when offline.
